gov_uk_dashboards/assets/mobile-nav.js

```diff
@@ -18,6 +18,21 @@ if (typeof attachEventToDash !== 'undefined') {
         mobileNav.style.display = isOpen ? 'none' : 'inline-block';
         isOpen = !isOpen;
     }, false)
+
+// attach click events that close the menu to links in the nav menu 
+var numberOfLinks = 8; // ideally get dynamically but document.querySelectorAll and trying to match id before attaching event to Dash didn't work.. 
```

Comment:
We can't hard code the number of links, as this component is in the package and so may be used across other dashboards.

Reply:
Agree - but the function does only attach events to links with a specific id of the format 'nav-bar-link-' + i + '-mobile' which has been set in LG repo - so we could always rename them to lg-'nav-bar-link-' + i + '-mobile' and then it would be obvious if someone was implementing in another package to give their links a different name. Just as a last resort -hope you have managed to find a solution to dynamically find the number of links.

Reply:
Hope you have managed to get working - another idea in case - might be able to create an API endpoint in our dash code the javascript code can query for the number of active dashboards so then not reliant on when the page loads